helm/cluster/templates/_helpers.tpl

```diff
@@ -151,4 +151,115 @@ Where `data` is the data to hash and `global` is the top level scope.
 {{- end -}}
 {{- end -}}
 {{- $registry -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{- define "cluster.internal.get-provider-integration-values" }}
+{{- $providerIntegration := dict }}
+{{- /* Case 1: template is called from the cluster chart itself */}}
+{{- if and (eq $.Chart.Name "cluster") $.Values.providerIntegration }}
+  {{- $providerIntegration = $.Values.providerIntegration }}
+{{- /* Case 2: template is called from a parent chart that passes Helm values to cluster chart in .Values.cluster */}}
+{{- else if ($.Values.cluster).providerIntegration }}
+  {{- $providerIntegration = $.Values.cluster.providerIntegration }}
+{{- end }}
+{{- /* Create $.GiantSwarm object where we put custom Giant Swarm vars */}}
+{{- if not $.GiantSwarm }}
+  {{- $_ := set $ "GiantSwarm" dict }}
+{{- end }}
+{{- /* Finally set "UseReleases" property that can be used in other templates */}}
+{{- $_ := set $.GiantSwarm "providerIntegration" $providerIntegration }}
+{{- end }}
+
+{{- define "cluster.internal.get-internal-values" }}
+{{- $internalValues := dict }}
+{{- /* Case 1: template is called from the cluster chart itself */}}
+{{- if and (eq $.Chart.Name "cluster") $.Values.internal }}
+  {{- $internalValues = $.Values.internal }}
+{{- /* Case 2: template is called from a parent chart that passes Helm values to cluster chart in .Values.cluster */}}
+{{- else if ($.Values.cluster).internal }}
+  {{- $internalValues = $.Values.cluster.internal }}
+{{- end }}
+{{- /* Create $.GiantSwarm.internal object */}}
+{{- if not $.GiantSwarm }}
+  {{- $_ := set $ "GiantSwarm" dict }}
+{{- end }}
+{{- /* Finally set "UseReleases" property that can be used in other templates */}}
+{{- $_ := set $.GiantSwarm "internal" $internalValues }}
+{{- end }}
+
+{{- define "cluster.internal.get-release-resource" }}
+{{- $_ := include "cluster.internal.get-internal-values" $ }}
+{{- $renderWithoutReleaseResource := ((($.GiantSwarm.internal).ephemeralConfiguration).offlineTesting).renderWithoutReleaseResource | default false }}
+{{- $releaseVersion := $.Values.global.release.version | trimPrefix "v" }}
+{{- $releaseVersion = printf "v%s" $releaseVersion }}
+{{- $release := lookup "release.giantswarm.io/v1alpha1" "Release" "" $releaseVersion }}
+{{- if $release }}
+  {{- $_ := set $.GiantSwarm "Release" $release }}
+{{ else if not $renderWithoutReleaseResource }}
+  {{- fail (printf "Release resource '%s' not found" $releaseVersion) }}
+{{- end }}
+{{- end }}
+
+{{- define "cluster.app.version" }}
+{{- $appVersion := "N/A" }}
+{{- $_ := (include "cluster.internal.get-release-resource" $) }}
+{{- if $.GiantSwarm.Release }}
+{{- range $_, $app := $.GiantSwarm.Release.spec.apps }}
+{{- if eq $app.name $.appName }}
+{{- $appVersion = $app.version }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{- $appVersion }}
+{{- end }}
+
+{{- define "cluster.component.version" }}
```

Comment:
Similar like for `cluster.app.version`, just for components (e.g. Kubernetes and OS version).